File: test/fixtures/node-axios/put_basic_auth_json_data.js
import axios from 'axios';

const response = await axios.put(
  'http://localhost:28139/test/_security',
  '{"admins":{"names":[], "roles":[]}, "readers":{"names":["joe"],"roles":[]}}',
  {
    headers: {
      'Content-Type': 'application/x-www-form-urlencoded'
    },
    auth: {
      username: 'admin',
      password: '123'
    }
  }
);

File: test/fixtures/ruby-httparty/put_basic_auth_json_data.rb
require 'httparty'

url = 'http://localhost:28139/test/_security'
auth = { username: 'admin', password: '123'}
headers = {
  'Content-Type': 'application/x-www-form-urlencoded',
}
body = '{"admins":{"names":[], "roles":[]}, "readers":{"names":["joe"],"roles":[]}}'
res = HTTParty.put(url, basic_auth: auth, headers: headers, body: body)
